block-node/messaging/src/main/java/org/hiero/block/server/messaging/impl/MessagingServiceImpl.java

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.block.server.messaging.impl;
+
+import com.hedera.hapi.block.BlockItemUnparsed;
+import com.lmax.disruptor.BatchEventProcessor;
+import com.lmax.disruptor.BatchEventProcessorBuilder;
+import com.lmax.disruptor.EventHandler;
+import com.lmax.disruptor.ExceptionHandler;
+import com.lmax.disruptor.RingBuffer;
+import com.lmax.disruptor.SequenceBarrier;
+import com.lmax.disruptor.SleepingWaitStrategy;
+import com.lmax.disruptor.dsl.Disruptor;
+import com.lmax.disruptor.dsl.ProducerType;
+import java.lang.System.Logger.Level;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ThreadFactory;
+import org.hiero.block.server.messaging.BlockItemHandler;
+import org.hiero.block.server.messaging.BlockNotification;
+import org.hiero.block.server.messaging.BlockNotificationHandler;
+import org.hiero.block.server.messaging.MessagingService;
+import org.hiero.block.server.messaging.NoBackPressureBlockItemHandler;
+
+/**
+ * Implementation of the MessagingService interface. It uses the LMAX Disruptor to handle block item batches and block
+ * notifications.
+ */
+public class MessagingServiceImpl implements MessagingService {
+
+    /** The size of the ring buffer used by the disruptor. */
+    public static final int RING_BUFFER_SIZE = 1024;
+
+    /** Logger for the messaging service. */
+    private static final System.Logger LOGGER = System.getLogger(MessagingServiceImpl.class.getName());
+
+    /**
+     * The thread factory used to create the virtual threads for the disruptor. Virtual threads are daemon threads by
+     * default.
+     */
+    public static final ThreadFactory VIRTUAL_THREAD_FACTORY = Thread.ofVirtual()
+            .name("messaging-service-handler", 0)
+            .uncaughtExceptionHandler((thread, throwable) -> LOGGER.log(
+                    Level.ERROR,
+                    "Uncaught exception in thread " + thread.getName() + ": " + throwable.getMessage(),
+                    throwable))
```

Comment:
Question, does this prevent the thread from throwing if it returns a Future and `get()` is called on that Future?

Also, we need to make sure the stack trace is in the log as follows:
```suggestion
            .uncaughtExceptionHandler((thread, throwable) -> LOGGER.log(
                    Level.ERROR,
                    "Uncaught exception in thread %s: %s.".formatted(thread.getName(), throwable.getMessage(),
                    throwable)) // Pass the throwable parameter to get a stack trace.
```

Reply:
these threads are not used with futures and my understanding is if we do not add a handler then the exception will just be silently lost

Reply:
This will happen if in any code that handles items or notifications from messaging services and throws an uncaught runtime exception. Feels like a complete nightmare to debug if not logged and is silently ignored.

Reply:
Makes sense, if never used in a way that something else will catch the exception, then we definitely need the handler.